include/zephyr/pm/state.h

```diff
@@ -156,6 +156,14 @@ struct pm_state_info {
 	uint32_t exit_latency_us;
 };
 
+/**
+ * @enum pm_state_direction Entry/exit indicator for state transitions
+ */
+enum pm_state_direction {
+	PM_STATE_ENTRY,
+	PM_STATE_EXIT,
+};
```

Comment:
Should't we have one for both ? If one needs to be notified before/after a state we can use only one callback, this can save space and timing iterating over the list

Reply:
In commit c4f8c1d6a63b68cdb91b77852ab5a82dcc218fd1 I moved/added additional `pm_state_notify()` calls to the edges of the respective state transitions: the exit of a low-power state is not exactly at the beginning of the active state, IMO:

https://github.com/zephyrproject-rtos/zephyr/blob/c4f8c1d6a63b68cdb91b77852ab5a82dcc218fd1/subsys/pm/pm.c#L152-L158

For my use case within a driver for reinit, I'd want to the callback to trigger before the `pm_state_exit_post_ops()` calls, while an application would probably want it to trigger after it. Hence the explicit distinction between low-power exit and active-state entry seen here.

Reply:
got that, saying that if you want to do something before/after `SOME_IDLE_STATE` you could use something like `PM_STATE_ENTRY_AND_EXIT`. For you case you use `PM_STATE_EXIT`, nothing changes. 

Note, that basically any state entry is another state leaving so what I am saying could be done with two notifiers, one when leaving active and other when leaving idle but having both directions option allow us to use only one calback.

Reply:
Ohh I see, it touches on your similar comment regarding the callback parameters. Sorry, misunderstood you. :)